sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -696,6 +706,24 @@ def connect(self):
                 http_proxy_port=http_proxy_port,
                 http_proxy_auth=http_proxy_auth,
             )
+        except WebSocketAddressException as exc:
+            raise AuthenticationException(
+                ErrorCondition.SocketError, # TODO: ClientError?
+                description="Failed to authenticate the connection due to exception: " + str(exc),
+                error=exc,
+            )
+        except (WebSocketTimeoutException, SSLError, WebSocketConnectionClosedException) as exc:    # type: ignore
+            self.close()
+            if isinstance(exc, WebSocketTimeoutException):
+                message = f'send timed out ({str(exc)})' 
+            elif isinstance(exc, SSLError):
+                message = f'send disconnected by SSL ({str(exc)})' 
```

Comment:
Minor nitpick: I hate to be that person... but can we start sentences with capital letters.... >_<